eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/batch/io/clustering_writer/RemappedVariantsClusteringWriterTest.java

```diff
@@ -300,24 +328,176 @@ public void test_variant_not_clustered_rs_id_exist() throws Exception {
+                     submittedVariantOperationEntity.getReason());
     }
 
-    private SubmittedVariantEntity getSubmittedVariantEntity(String assembly, String project, long start, Long rs, long ss, String remappedFrom) {
-        SubmittedVariant submittedClustered = new SubmittedVariant(assembly, TAXONOMY, project, CONTIG, start, T, A, rs);
+    /**
+     * This test is for testing a specific scenario in clustering where an already clustered variant is remapped but the
+     * remapped variant is at a different location or a different variant type.
+     *
+     * SubmittedVariantEntity                                                           ClusteredVariantEntity
+     * SS	RS	ASM	    STUDY	CONTIG	POS	    REF	ALT                    RS	HASH                ASM	    POS	    CONTIG	TYPE
+     * 501	306	ASM1	PRJEB1	Chr1	1000		TT  (org)              306  ASM1_Chr1_1000_INS	ASM1	1000	Chr1	INS
+     * 500	306	ASM1	PRJEB1	Chr1	1000	A	AT  (org)
+     * 501	306	ASM2	PRJEB1	Chr1	1500		TT  (remapped)
+     * 500	306	ASM2	PRJEB1	Chr1	1000		T   (remapped)
+     *
+     * SS id 501 and 500 has same RS because of remapping, but they are now at different positions and can't have same RS id.
+     * RS split needs to be done here, with one retainine old RS id while the other should get a new RS id
+     *
+     * After Clustering
+     *
+     * SubmittedVariantEntity                                                           ClusteredVariantEntity
+     * SS	RS	ASM	    STUDY	CONTIG	POS	    REF	ALT                    RS	HASH                ASM	    POS	    CONTIG	TYPE
+     * 501	306	ASM1	PRJEB1	Chr1	1000		TT  (org)              306  ASM1_Chr1_1000_INS	ASM1	1000	Chr1	INS
+     * 500	306	ASM1	PRJEB1	Chr1	1000	A	AT  (org)              306  ASM2_Chr1_1000_INS	ASM2	1000	Chr1	INS
+     * 501	400	ASM2	PRJEB1	Chr1	1500		TT  (remapped)         400  ASM2_Chr1_1500_INS	ASM2	1500	Chr1	INS
+     * 500	306	ASM2	PRJEB1	Chr1	1000		T   (remapped)
+     */
+    @Test
+    @DirtiesContext
+    public void test_clustered_variant_remapped_RS_split() throws Exception {
+        //original submitted variants
+        SubmittedVariantEntity submittedVariantEntity1 = getSubmittedVariantEntity(ASM_1, PROJECT_1, 1000, 3000000006L,
+                                                                                   5000000001L, "", "TT", NOT_REMAPPED);
+        mongoTemplate.insert(submittedVariantEntity1, SUBMITTED_VARIANT_COLLECTION);
+        SubmittedVariantEntity submittedVariantEntity2 = getSubmittedVariantEntity(ASM_1, PROJECT_1, 1000, 3000000006L,
+                                                                                   5000000000L, "A", "AT",
+                                                                                   NOT_REMAPPED);
+        mongoTemplate.insert(submittedVariantEntity2, SUBMITTED_VARIANT_COLLECTION);
+        SubmittedVariantEntity submittedVariantEntity3 = getSubmittedVariantEntity(ASM_1, PROJECT_2, 3000, 3000000007L,
```

Comment:
I see variants `submittedVariantEntity3` and `submittedVariantEntity4` are not part of the simple case we are trying to test and do not appear in the description. 

Can we remove them so this test is as simple as possible? 

If there are some extra scenarios where those variants should be involved you could create another (more complex) test case but having the simplest one will help understand this scenario better

Reply:
removed